proxy/src/ctx/transport.rs

```diff
@@ -22,11 +25,12 @@ pub struct Server {
 }
 
 /// Identifies a connection from the proxy to another process.
-#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+#[derive(Clone, Debug)]
 pub struct Client {
     pub proxy: Arc<ctx::Proxy>,
     pub remote: SocketAddr,
     pub protocol: Protocol,
+    pub dst_labels: Option<Watch<Option<DstLabels>>>,
```

Comment:
The repetition of this fairly unfortunate type (of which I'm the original perpetrator) makes me wonder if maybe we should just move the inner `Option` into `DstLabels` instead --- but that seems like a change to make in a separate PR.

Reply:
And/or`control::discovery` could export a `pub type DstLabelsWatch = Option<Watch<Option<DstLabels>>>`or something...